code-challenges/Data-Structures/tree/tree.js

```diff
@@ -11,6 +11,18 @@ class BinaryTree {
     this.root = null
   }
 
+  matchTree (arr) {
+    const output = []
+    const matchTraverse = (Node, arr) => {
+      if (!Node) return
+      if (arr.includes(Node.value)) output.push(Node.value)
```

Comment:
Hi Eugene,
Think about the time complexity of this line. Can we do better by using some data structure other than array?